File: app/src/main/java/com/semisonfire/cloudgallery/data/local/LocalDatabase.java
package com.semisonfire.cloudgallery.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.semisonfire.cloudgallery.data.local.dao.PhotoDao;
import com.semisonfire.cloudgallery.data.model.Photo;

@Database(entities = Photo.class, version = 1, exportSchema = false)
public abstract class LocalDatabase extends RoomDatabase {

    public abstract PhotoDao getPhotoDao();

    private static LocalDatabase instance;

    public static LocalDatabase getInstance(Context context) {

        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), LocalDatabase.class, "cloud-gallery.db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }

}

File: app/src/main/java/com/semisonfire/cloudgallery/data/local/LocalDatabase.kt
package com.semisonfire.cloudgallery.data.local

import androidx.room.Database
import androidx.room.RoomDatabase

import com.semisonfire.cloudgallery.data.local.dao.PhotoDao
import com.semisonfire.cloudgallery.data.model.Photo

@Database(
    entities = [
        Photo::class
    ],
    version = LocalDatabase.DATABASE_VERSION,
    exportSchema = false
)
abstract class LocalDatabase : RoomDatabase() {

    /*Dao*/
    abstract val photoDao: PhotoDao

    companion object {
        internal const val DATABASE_VERSION = 2
    }

    /*Migrations*/
}